pkg/sql/planner.go

```diff
@@ -72,6 +72,8 @@ type extendedEvalContext struct {
 
 	Jobs *jobsCollection
 
+	JobsCache *map[sqlbase.ID]*jobs.Job
```

Comment:
Does this need to be a pointer? I thought just copying the map by value took a reference

Reply:
Is it also possible to comment what this is aliasing (as well as the Jobs) alias above? I've found it difficult in the past to figure out how the `extraTxnState` was getting modified.

Reply:
I think it actually does have to be a pointer, I don't think maps are pass by reference in go.
Just tried making it not a pointer and it doesn't work.

Also by aliasing do you mean how Jobs/JobsCache refers to values in extraTxnState?

Reply:
Something is wonky, because maps are surely pass by reference -- you can pass a map into a function which fills up the map for you.

> Also by aliasing do you mean how Jobs/JobsCache refers to values in extraTxnState?

Yeah exactly.

Reply:
> Something is wonky, because maps are surely pass by reference -- you can pass a map into a function which fills up the map for you.
> 
> > Also by aliasing do you mean how Jobs/JobsCache refers to values in extraTxnState?
> 
> Yeah exactly.

Wait yeah they're definitely pass by reference... I'm not sure why the pointer is necessary here, hmm

Reply:
Nvm I was being dumb, the pointer is necessary because we actually want the map to be reset by the call to resetExtraTxnState since it resets the cache. Without the pointer the cache doesn't get cleared when a txn is committed